Assets/Scripts/AI/StateMachine/EnemyStates/EnemyShootingState.cs

```diff
@@ -30,6 +29,9 @@ public override EnemyStateManager.EnemyState GetNextState()
         {
             if (CurrentGun.BulletsRemaining <= 0)
                 return EnemyStateManager.EnemyState.Reloading;
+            
+            if (_defenseStats.NormalizedHealth <= 0.3f)
```

Comment:
this is causing infinite loop since in hiding it returns back to previous state. Should add an and condition that previousState != Hiding && ..

Reply:
Ok I'll try to add a minimum time after which only it can go from shooting to hiding